### ISSUE TITLE
Crash while switching tabs quickly

### ISSUE THREAD

**Issue by shuvashish76** [MuntashirAkon/AppManager#1190]
### Please check before submitting an issue

- [X] I know what my device, OS and App Manager versions are
- [X] I know how to take logs
- [ ] I know how to reproduce the issue which may not be specific to my device

### Describe the bug

AM crashed while switching Appinfo/Component/Permission tabs quickly.
(1 time crash)

### Logs

```
java.lang.IndexOutOfBoundsException: Inconsistency detected. Invalid view holder adapter positionViewHolder{4bb3881 position=61 id=-1, oldPos=3, pLpos:3 scrap [attachedScrap] tmpDetached no parent} io.github.muntashirakon.widget.RecyclerView{3a59a5a VFED..... ......ID 0,0-1080,2051 #7f0902bb app:id/scrollView}, adapter:io.github.muntashirakon.AppManager.details.AppDetailsPermissionsFragment$AppDetailsRecyclerAdapter@5e691d5, layout:androidx.recyclerview.widget.LinearLayoutManager@3592dea, context:io.github.muntashirakon.AppManager.details.AppDetailsActivity@dd63496
    at androidx.recyclerview.widget.RecyclerView$Recycler.validateViewHolderForOffsetPosition(RecyclerView.java:6156)
    at androidx.recyclerview.widget.RecyclerView$Recycler.tryGetViewHolderForPositionByDeadline(RecyclerView.java:6339)
    at androidx.recyclerview.widget.RecyclerView$Recycler.getViewForPosition(RecyclerView.java:6300)
    at androidx.recyclerview.widget.RecyclerView$Recycler.getViewForPosition(RecyclerView.java:6296)
    at androidx.recyclerview.widget.LinearLayoutManager$LayoutState.next(LinearLayoutManager.java:2330)
    at androidx.recyclerview.widget.LinearLayoutManager.layoutChunk(LinearLayoutManager.java:1631)
    at androidx.recyclerview.widget.LinearLayoutManager.fill(LinearLayoutManager.java:1591)
    at androidx.recyclerview.widget.LinearLayoutManager.onLayoutChildren(LinearLayoutManager.java:668)
    at androidx.recyclerview.widget.RecyclerView.dispatchLayoutStep1(RecyclerView.java:4255)
    at androidx.recyclerview.widget.RecyclerView.dispatchLayout(RecyclerView.java:4010)
    at androidx.recyclerview.widget.RecyclerView.onLayout(RecyclerView.java:4578)
    at android.view.View.layout(View.java:23265)
    at android.view.ViewGroup.layout(ViewGroup.java:6415)
    at android.widget.FrameLayout.layoutChildren(FrameLayout.java:332)
    at android.widget.FrameLayout.onLayout(FrameLayout.java:270)
    at android.view.View.layout(View.java:23265)
    at android.view.ViewGroup.layout(ViewGroup.java:6415)
    at android.widget.RelativeLayout.onLayout(RelativeLayout.java:1103)
    at android.view.View.layout(View.java:23265)
    at android.view.ViewGroup.layout(ViewGroup.java:6415)
    at androidx.swiperefreshlayout.widget.SwipeRefreshLayout.onLayout(SwipeRefreshLayout.java:688)
    at android.view.View.layout(View.java:23265)
    at android.view.ViewGroup.layout(ViewGroup.java:6415)
    at android.widget.FrameLayout.layoutChildren(FrameLayout.java:332)
    at android.widget.FrameLayout.onLayout(FrameLayout.java:270)
    at android.view.View.layout(View.java:23265)
    at android.view.ViewGroup.layout(ViewGroup.java:6415)
    at androidx.recyclerview.widget.RecyclerView$LayoutManager.layoutDecoratedWithMargins(RecyclerView.java:9878)
    at androidx.recyclerview.widget.LinearLayoutManager.layoutChunk(LinearLayoutManager.java:1689)
    at androidx.recyclerview.widget.LinearLayoutManager.fill(LinearLayoutManager.java:1591)
    at androidx.recyclerview.widget.LinearLayoutManager.onLayoutChildren(LinearLayoutManager.java:668)
    at androidx.recyclerview.widget.RecyclerView.dispatchLayoutStep2(RecyclerView.java:4309)
    at androidx.recyclerview.widget.RecyclerView.dispatchLayout(RecyclerView.java:4012)
    at androidx.recyclerview.widget.RecyclerView.onLayout(RecyclerView.java:4578)
    at android.view.View.layout(View.java:23265)
    at android.view.ViewGroup.layout(ViewGroup.java:6415)
    at androidx.viewpager2.widget.ViewPager2.onLayout(ViewPager2.java:527)
    at android.view.View.layout(View.java:23265)
    at android.view.ViewGroup.layout(ViewGroup.java:6415)
    at com.google.android.material.appbar.HeaderScrollingViewBehavior.layoutChild(HeaderScrollingViewBehavior.java:149)
    at com.google.android.material.appbar.ViewOffsetBehavior.onLayoutChild(ViewOffsetBehavior.java:43)
    at com.google.android.material.appbar.AppBarLayout$ScrollingViewBehavior.onLayoutChild(AppBarLayout.java:2376)
    at androidx.coordinatorlayout.widget.CoordinatorLayout.onLayout(CoordinatorLayout.java:918)
    at android.view.View.layout(View.java:23265)
    at android.view.ViewGroup.layout(ViewGroup.java:6415)
    at android.widget.FrameLayout.layoutChildren(FrameLayout.java:332)
    at android.widget.FrameLayout.onLayout(FrameLayout.java:270)
    at android.view.View.layout(View.java:23265)
    at android.view.ViewGroup.layout(ViewGroup.java:6415)
    at android.widget.LinearLayout.setChildFrame(LinearLayout.java:1829)
    at android.widget.LinearLayout.layoutVertical(LinearLayout.java:1673)
    at android.widget.LinearLayout.onLayout(LinearLayout.java:1582)
    at android.view.View.layout(View.java:23265)
    at android.view.ViewGroup.layout(ViewGroup.java:6415)
    at android.widget.FrameLayout.layoutChildren(FrameLayout.java:332)
    at android.widget.FrameLayout.onLayout(FrameLayout.java:270)
    at android.view.View.layout(View.java:23265)
    at android.view.ViewGroup.layout(ViewGroup.java:6415)
    at android.widget.LinearLayout.setChildFrame(LinearLayout.java:1829)
    at android.widget.LinearLayout.layoutVertical(LinearLayout.java:1673)
    at android.widget.LinearLayout.onLayout(LinearLayout.java:1582)
    at android.view.View.layout(View.java:23265)
    at android.view.ViewGroup.layout(ViewGroup.java:6415)
    at android.widget.FrameLayout.layoutChildren(FrameLayout.java:332)
    at android.widget.FrameLayout.onLayout(FrameLayout.java:270)
    at com.android.internal.policy.DecorView.onLayout(DecorView.java:831)
    at android.view.View.layout(View.java:23265)
    at android.view.ViewGroup.layout(ViewGroup.java:6415)
    at android.view.ViewRootImpl.performLayout(ViewRootImpl.java:3869)
    at android.view.ViewRootImpl.performTraversals(ViewRootImpl.java:3328)
    at android.view.ViewRootImpl.doTraversal(ViewRootImpl.java:2282)
    at android.view.ViewRootImpl$TraversalRunnable.run(ViewRootImpl.java:8986)
    at android.view.Choreographer$CallbackRecord.run(Choreographer.java:1037)
    at android.view.Choreographer.doCallbacks(Choreographer.java:845)
    at android.view.Choreographer.doFrame(Choreographer.java:780)
    at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:1022)
    at android.os.Handler.handleCallback(Handler.java:978)
    at android.os.Handler.dispatchMessage(Handler.java:104)
    at android.os.Looper.loopOnce(Looper.java:238)
    at android.os.Looper.loop(Looper.java:357)
    at android.app.ActivityThread.main(ActivityThread.java:8090)
    at java.lang.reflect.Method.invoke(Native Method)
    at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1026)
```

### Device info

App version: 4.0.0-alpha01-DEBUG
App version code: 434
Android build version: 8049b-fa5cf
Android release version: 12
Android SDK version: 31
Android build ID: S2RUBS32.51-15-9-7
Device brand: motorola
Device manufacturer: motorola
Device name: corfur
Device model: moto g71 5G
Device product name: corfur_g
Device hardware name: qcom
ABIs: [arm64-v8a, armeabi-v7a, armeabi]
ABIs (32bit): [armeabi-v7a, armeabi]
ABIs (64bit): [arm64-v8a]
System language: en-IN
In-App Language: auto
Mode: root
Inferred Mode: Root

**Comment by shuvashish76** [comment]
Slightly different but similar crash.
Reproducible when
1. Turn of "Instant component blocking"
2. Turn on/off component switches very fast
3. AM crashes for v4.0.0-alpha02 but not for v3.1.2

"_Couldn't enable the requested app op_" message shown if you change app op status quickly but no crashes yet.
<details><summary>Logs</summary>
<p>

```
java.lang.IndexOutOfBoundsException: Inconsistency detected. Invalid item position 6(offset:6).state:7 io.github.muntashirakon.widget.RecyclerView{88f440a VFED..... ......ID 0,0-1080,2051 #7f0902bb app:id/scrollView}, adapter:io.github.muntashirakon.AppManager.details.AppDetailsComponentsFragment$AppDetailsRecyclerAdapter@7f0f8a7, layout:androidx.recyclerview.widget.LinearLayoutManager@586f154, context:io.github.muntashirakon.AppManager.details.AppDetailsActivity@8c1f2a3
    at androidx.recyclerview.widget.RecyclerView$Recycler.tryGetViewHolderForPositionByDeadline(RecyclerView.java:6364)
    at androidx.recyclerview.widget.RecyclerView$Recycler.getViewForPosition(RecyclerView.java:6300)
    at androidx.recyclerview.widget.RecyclerView$Recycler.getViewForPosition(RecyclerView.java:6296)
    at androidx.recyclerview.widget.LinearLayoutManager$LayoutState.next(LinearLayoutManager.java:2330)
    at androidx.recyclerview.widget.LinearLayoutManager.layoutChunk(LinearLayoutManager.java:1631)
    at androidx.recyclerview.widget.LinearLayoutManager.fill(LinearLayoutManager.java:1591)
    at androidx.recyclerview.widget.LinearLayoutManager.onLayoutChildren(LinearLayoutManager.java:668)
    at androidx.recyclerview.widget.RecyclerView.dispatchLayoutStep1(RecyclerView.java:4255)
    at androidx.recyclerview.widget.RecyclerView.dispatchLayout(RecyclerView.java:4010)
    at androidx.recyclerview.widget.RecyclerView.onLayout(RecyclerView.java:4578)
    at android.view.View.layout(View.java:23265)
    at android.view.ViewGroup.layout(ViewGroup.java:6415)
    at android.widget.FrameLayout.layoutChildren(FrameLayout.java:332)
    at android.widget.FrameLayout.onLayout(FrameLayout.java:270)
    at android.view.View.layout(View.java:23265)
    at android.view.ViewGroup.layout(ViewGroup.java:6415)
    at android.widget.RelativeLayout.onLayout(RelativeLayout.java:1103)
    at android.view.View.layout(View.java:23265)
    at android.view.ViewGroup.layout(ViewGroup.java:6415)
    at androidx.swiperefreshlayout.widget.SwipeRefreshLayout.onLayout(SwipeRefreshLayout.java:688)
    at android.view.View.layout(View.java:23265)
    at android.view.ViewGroup.layout(ViewGroup.java:6415)
    at android.widget.FrameLayout.layoutChildren(FrameLayout.java:332)
    at android.widget.FrameLayout.onLayout(FrameLayout.java:270)
    at android.view.View.layout(View.java:23265)
    at android.view.ViewGroup.layout(ViewGroup.java:6415)
    at androidx.recyclerview.widget.RecyclerView$LayoutManager.layoutDecoratedWithMargins(RecyclerView.java:9878)
    at androidx.recyclerview.widget.LinearLayoutManager.layoutChunk(LinearLayoutManager.java:1689)
    at androidx.recyclerview.widget.LinearLayoutManager.fill(LinearLayoutManager.java:1591)
    at androidx.recyclerview.widget.LinearLayoutManager.onLayoutChildren(LinearLayoutManager.java:678)
    at androidx.recyclerview.widget.RecyclerView.dispatchLayoutStep2(RecyclerView.java:4309)
    at androidx.recyclerview.widget.RecyclerView.dispatchLayout(RecyclerView.java:4012)
    at androidx.recyclerview.widget.RecyclerView.onLayout(RecyclerView.java:4578)
    at android.view.View.layout(View.java:23265)
    at android.view.ViewGroup.layout(ViewGroup.java:6415)
    at androidx.viewpager2.widget.ViewPager2.onLayout(ViewPager2.java:527)
    at android.view.View.layout(View.java:23265)
    at android.view.ViewGroup.layout(ViewGroup.java:6415)
    at com.google.android.material.appbar.HeaderScrollingViewBehavior.layoutChild(HeaderScrollingViewBehavior.java:149)
    at com.google.android.material.appbar.ViewOffsetBehavior.onLayoutChild(ViewOffsetBehavior.java:43)
    at com.google.android.material.appbar.AppBarLayout$ScrollingViewBehavior.onLayoutChild(AppBarLayout.java:2376)
    at androidx.coordinatorlayout.widget.CoordinatorLayout.onLayout(CoordinatorLayout.java:918)
    at android.view.View.layout(View.java:23265)
    at android.view.ViewGroup.layout(ViewGroup.java:6415)
    at android.widget.FrameLayout.layoutChildren(FrameLayout.java:332)
    at android.widget.FrameLayout.onLayout(FrameLayout.java:270)
    at android.view.View.layout(View.java:23265)
    at android.view.ViewGroup.layout(ViewGroup.java:6415)
    at android.widget.LinearLayout.setChildFrame(LinearLayout.java:1829)
    at android.widget.LinearLayout.layoutVertical(LinearLayout.java:1673)
    at android.widget.LinearLayout.onLayout(LinearLayout.java:1582)
    at android.view.View.layout(View.java:23265)
    at android.view.ViewGroup.layout(ViewGroup.java:6415)
    at android.widget.FrameLayout.layoutChildren(FrameLayout.java:332)
    at android.widget.FrameLayout.onLayout(FrameLayout.java:270)
    at android.view.View.layout(View.java:23265)
    at android.view.ViewGroup.layout(ViewGroup.java:6415)
    at android.widget.LinearLayout.setChildFrame(LinearLayout.java:1829)
    at android.widget.LinearLayout.layoutVertical(LinearLayout.java:1673)
    at android.widget.LinearLayout.onLayout(LinearLayout.java:1582)
    at android.view.View.layout(View.java:23265)
    at android.view.ViewGroup.layout(ViewGroup.java:6415)
    at android.widget.FrameLayout.layoutChildren(FrameLayout.java:332)
    at android.widget.FrameLayout.onLayout(FrameLayout.java:270)
    at com.android.internal.policy.DecorView.onLayout(DecorView.java:831)
    at android.view.View.layout(View.java:23265)
    at android.view.ViewGroup.layout(ViewGroup.java:6415)
    at android.view.ViewRootImpl.performLayout(ViewRootImpl.java:3869)
    at android.view.ViewRootImpl.performTraversals(ViewRootImpl.java:3328)
    at android.view.ViewRootImpl.doTraversal(ViewRootImpl.java:2282)
    at android.view.ViewRootImpl$TraversalRunnable.run(ViewRootImpl.java:8986)
    at android.view.Choreographer$CallbackRecord.run(Choreographer.java:1037)
    at android.view.Choreographer.doCallbacks(Choreographer.java:845)
    at android.view.Choreographer.doFrame(Choreographer.java:780)
    at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:1022)
    at android.os.Handler.handleCallback(Handler.java:978)
    at android.os.Handler.dispatchMessage(Handler.java:104)
    at android.os.Looper.loopOnce(Looper.java:238)
    at android.os.Looper.loop(Looper.java:357)
    at android.app.ActivityThread.main(ActivityThread.java:8090)
    at java.lang.reflect.Method.invoke(Native Method)
    at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1026)

Device Info:
App version: 4.0.0-alpha02
App version code: 435
Android build version: 8049b-fa5cf
Android release version: 12
Android SDK version: 31
Android build ID: S2RUBS32.51-15-9-7
Device brand: motorola
Device manufacturer: motorola
Device name: corfur
Device model: moto g71 5G
Device product name: corfur_g
Device hardware name: qcom
ABIs: [arm64-v8a, armeabi-v7a, armeabi]
ABIs (32bit): [armeabi-v7a, armeabi]
ABIs (64bit): [arm64-v8a]
System language: en-IN
In-App Language: auto
Mode: root
Inferred Mode: Root
```

</p>
</details> 

Note: I'm sure in the previous crash it happened due to switching tabs not changing component status. Still not reproducible.

**Comment by MuntashirAkon** [comment]
Fixed via d8ba865c38a74cf1a37d79996678fb147dd605cb
